File: nan21.dnet.module.pj/nan21.dnet.module.pj.presenter/src/main/java/net/nan21/dnet/module/pj/md/ds/model/ItemLovDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2010 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.pj.md.ds.model;

import net.nan21.dnet.core.api.model.IModelWithClientId;
import net.nan21.dnet.core.api.model.IModelWithId;
import net.nan21.dnet.core.presenter.model.AbstractDsModel;

import net.nan21.dnet.module.pj.md.domain.entity.Item;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;

@Ds(entity = Item.class)
public class ItemLovDs extends AbstractDsModel<Item> implements IModelWithId,
        IModelWithClientId {

    public static final String fID = "id";
    public static final String fCLIENTID = "clientId";
    public static final String fSUMMARY = "summary";

    @DsField()
    private Long id;

    @DsField()
    private Long clientId;

    @DsField()
    private String summary;

    public ItemLovDs() {
        super();
    }

    public ItemLovDs(Item e) {
        super(e);
    }

    public Long getId() {
        return this.id;
    }

    public void setId(Object id) {
        this.id = this._asLong_(id);

    }

    public Long getClientId() {
        return this.clientId;
    }

    public void setClientId(Long clientId) {
        this.clientId = clientId;
    }

    public String getSummary() {
        return this.summary;
    }

    public void setSummary(String summary) {
        this.summary = summary;
    }

}

File: nan21.dnet.module.pj/nan21.dnet.module.pj.ui.extjs/src/main/webapp/src/ds/ItemLovDs.js
 
   	 
Ext.define("net.nan21.dnet.module.pj.md.ds.model.ItemLovDs" , {
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id",type:"int", useNull:true },
		{name:"clientId",type:"int", useNull:true },
		{name:"summary",type:"string" }
	]
});

Ext.define("net.nan21.dnet.module.pj.md.ds.model.ItemLovDsFilter" , {
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id",type:"int", useNull:true },
		{name:"clientId",type:"int", useNull:true },
		{name:"summary",type:"string" }
	] 
});
	
Ext.define("net.nan21.dnet.module.pj.md.ds.param.ItemLovDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
}); 
 